diesel/src/query_builder/mod.rs

```diff
@@ -25,6 +25,7 @@ pub mod nodes;
 mod offset_clause;
 mod order_clause;
 mod returning_clause;
+mod select_by_statement;
```

Comment:
I would rather like to have that as module in `select_statement`

Reply:
Why? I think `select_by_statement` is independent with `select_statement`, there is not any restriction that inner `Stmt` should be a `select_statement`.
And I think it is also large enough to have its own folder. If as a module, maybe we would have `diesel/src/query_builder/select_statement/select_by_statement/dsl_impls.rs`? This is unlike `BoxedSelectStatement` which is the same thing as `SelectStatement`.
If you still think subdir is better I could transfer it.

Reply:
Just a note: I don't see why we couldn't have `InsertStatement`, `UpdateStatement` and even `DeleteStatement` behind the scenes thanks to `returning` clause. Or shall we create `InsertReturningByStatment`, `UpdateReturningByStatement` and `DeleteReturningByStatement`?